frontend/components/ClubEditPage/ClubEditCard.tsx

```diff
@@ -871,9 +916,36 @@ export default function ClubEditCard({
     : creationDefaults
 
   return (
-    <Formik initialValues={initialValues} onSubmit={submit} enableReinitialize>
-      {({ dirty, isSubmitting }) => (
+    <Formik
+      initialValues={initialValues}
+      onSubmit={(values, actions) =>
+        submit({ ...values, emailOverride: false }, actions)
+      }
+      enableReinitialize
+      validate={(values) => {
+        const errors: { email?: string } = {}
+        if (values.email.includes('upenn.edu') && !emailModal) {
+          showEmailModal(true)
+          errors.email = 'Please confirm your email'
```

Comment:
I'm not sure if this error is ever shown (when I tested it)

Reply:
I don't believe it is. For validation to work in FormiK, a dictionary of errors need to be returned. If the dictionary isn't returned, onSubmit will execute. In case we want to continue adding to the errors dictionary with other errors, I created the errors object. I'm sure there is a way to display the errors, but I didn't think it is necessary, as the modal does this.